test/test_type_promotion.py

```diff
@@ -183,12 +183,14 @@ def test_bfloat16(self, device):
             self.assertEqual(bf + scalar, scalar + bf)
 
         # with tensor
-        for dtype in all_types_and_complex_and(torch.half, torch.bfloat16, torch.bool):
+        for dtype in all_types_and_complex_and(torch.chalf, torch.half, torch.bfloat16, torch.bool):
```

Comment:
annoying because this entire test fails because `complex32` is not supported on xla. @JackCaoG is it possible to support it on XLA? 

If not, we should add a small test separately for testing bfloat16 and complex32 and run it only on native devices

Reply:
unfortunately complex32 is not supported by xla compiler.